core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -325,6 +325,68 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  def setBillingProjectSpendConfiguration(billingProjectName: RawlsBillingProjectName, spendReportConfiguration: BillingProjectSpendConfiguration): Future[Int] = {
+
+    val datasetName = spendReportConfiguration.datasetName
+    val datasetGoogleProject = spendReportConfiguration.datasetGoogleProject
+
+    validateBigQueryDatasetName(datasetName)
+    validateGoogleProjectName(datasetGoogleProject.value)
+
+    requireProjectAction(billingProjectName, SamBillingProjectActions.alterSpendReportConfiguration) {
+      val bqService = bqServiceFactory.getServiceFromJson(bigQueryCredentialJson, GoogleProject(billingProjectName.value))
+
+      for {
+        //Get the dataset to validate that it exists and that we have permission to see it
+        _ <- bqService.use(_.getDataset(datasetGoogleProject, datasetName)).unsafeToFuture().map {
+          case None => throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, s"The dataset $datasetName could not be found."))
+          case dataset => dataset
+        }
+
+        billingAccountId <- dataSource.inTransaction { dataAccess =>
```

Comment:
In develop, we get the `billingAccountId` from Google because in a pre-PPW world we cannot rely on the value stored in the Rawls DB. Once PPW is live, the billing account stored in the Rawls DB will be accurate because all billing account updates will have to go through Rawls (we're taking away the permission for the user to update it directly). `setBillingProjectSpendConfiguration` is untouched other than this database query that I substituted for the call to Google that used to be here.